play/src/main/scala/com/managementapp/authentication/controllers/AuthenticationController.scala

```diff
@@ -0,0 +1,32 @@
+package com.managementapp.authentication.controllers
+
+import play.api.mvc.{Action, AnyContent, Controller}
+import com.managementapp.common.actions.AuthenticatedAction
+import com.google.inject.Inject
+import com.managementapp.common.util.ProtobufUtil
+import com.managementapp.transfers.Authentication.AccountDTO
+import com.managementapp.authentication.services.AuthenticationService
+import com.managementapp.common.services.UserService
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class AuthenticationController @Inject() (authService: AuthenticationService, userService: UserService) extends Controller with ProtobufUtil {
+  def sampleActionThatRequiresAuth: Action[AnyContent] = AuthenticatedAction(authService) { request =>
+    //this route requires login first
+    Ok("Logged in user is: " + request.user.email)
+  }
+
+  def createAccount: Action[AnyContent] = Action {
+    request =>
+      val account = parseJSONProtocolBuffer[AccountDTO](AccountDTO.newBuilder, request)
```

Comment:
I'm wondering is this a typical use case for protobuf? I imagine how we would use protobuf is that when front-end sending the request, the data in the request body has been transformed to ProtoBuf format, and in the back-end we can directly transform it to the server language (ie scala/java) and use it. 
Right now, it looks a bit overkilled to me as if we want to send messages as Json, it's more straightforward to parse json directly.
I'm not familiar with protobuf, correct me if I'm wrong.
just wondering when should we use it in the case of building a web api?